frontend/src/hooks/useAuthenticatedUser.js

```diff
@@ -19,9 +19,7 @@ export function useAuthenticatedUser() {
       }
     }
 
-    if (user) {
-      getUserData()
-    }
+    user ? getUserData() : setUserData(null)
```

Comment:
Any reason we want to use `null` instead of `undefined` here? What was wrong with the old behavior of leaving it as `undefined`?

Reply:
Oh yeah I added this just to explicitly set the userMetaData to something falsey whenever the auth state (`[user]`) was changed. This was done so that whenever the user clicked logout, the userMetaData would update to reflect the logout change in `[user]`, since the meta data is being used for posting the booking to Firestore.

Reply:
Gotcha (good catch!). We should probably stick to `userData` being either exactly `undefined` or the user, or exactly `null` or the user (i.e., not default to `undefined` and then later have the potential to be `null`).

I think either is fine, I'd probably lean to `undefined` though.

Reply:
Sounds good to me, I've made another commit to set the userMetaData to be undefined instead of null on sign out 👍